tls/s2n_client_hello.c

```diff
@@ -118,6 +122,16 @@ int s2n_client_hello_free(struct s2n_client_hello *client_hello) {
     return 0;
 }
 
+int s2n_client_hello_free_parsed_extensions(struct s2n_client_hello *client_hello)
+{
+    notnull_check(client_hello);
+    if (client_hello->parsed_extensions != NULL) {
+        s2n_array_free(client_hello->parsed_extensions);
+    }
+    client_hello->parsed_extensions = NULL;
```

Comment:
nitpick: required only if not null, could be moved in if not null scope

Reply:
Sure moved.